rust/azure_iot_operations_mqtt/src/session/reconnect_policy.rs

```diff
@@ -10,9 +10,13 @@ use rand::Rng;
 use crate::control_packet::Disconnect;
 use crate::error::{ConnectError, ProtocolError};
 
-pub enum ConnectionLoss {
+/// Reason for connection loss.
+pub enum ConnectionLossReason {
+    /// Disconnected by server with DISCONNECT packet.
     DisconnectByServer(Disconnect),
+    /// Disconnected due to an I/O error.
     IoError(std::io::Error),
+    /// Disconnected due to a protocol error.
```

Comment:
or something like that, not sure if committed is too aggressive
```suggestion
    /// Disconnected due to a protocol error committed by the server.
```